_posts/2020-08-18-scaffolding-static-site-with-cdk.md

```diff
@@ -0,0 +1,243 @@
+
+```typescript
+  private getRoute53Records(
+    distribution: IDistribution,
+    hostedZone: IHostedZone,
+    url: string
+  ) {
+    new ARecord(this, "Alias", {
+      zone: hostedZone,
+      recordName: url,
+      target: RecordTarget.fromAlias(new CloudFrontTarget(distribution)),
+    });
+
+    new HttpsRedirect(this, "Redirect", {
+      zone: hostedZone,
+      recordNames: [`www.${url}`],
+      targetDomain: url,
+    });
+  }
+```
+
+We also need to inform CloudFront that we are using an SSL certificate to serve the content so we make the following changes to our `getCloudFrontDistribution` function by passing the certificate we created above as one of the parameters.
+
+```typescript
+const viewerCertificate = ViewerCertificate.fromAcmCertificate(certificate, {
+  aliases: [url],
+});
+
+const distribution = new CloudFrontWebDistribution(
+  this,
+  "WebsiteDistribution",
+  {
+    originConfigs: [originConfig],
+    errorConfigurations,
+    viewerCertificate,
+  }
+);
+```
+
+Putting it all together, we can now change our `constructor` to include the certificate and Hosted Zone as below.
+
+```typescript
+const bucketName = "some-bucket-name";
+const url = "some-url.com";
+
+const bucket = this.getS3Bucket(bucketName);
+const { certificate, hostedZone } = this.getRoute53HostedZone(url);
+const { distribution } = this.getCloudFrontDistribution(
+  bucket,
+  certificate,
+  url
+);
+this.getRoute53Records(distribution, hostedZone, url);
+
+new CfnOutput(this, "URL", {
+  description: "The URL of the site",
+  value: hostedZone.zoneName,
+});
+```
+
+Once we run `cdk synth` and `cdk deploy`, we should see our custom domain in the output. Navigating to it should show our site content along with the SSL certificate.
+
+So there you have it, a reusable template for deploying static sites using AWS CDK. You can view the [full template on GitHub](https://gist.github.com/timveletta/4e751f4add532ed246642d061ad3fcb9) and if you have any questions feel free to contact me on Twitter ([@TimVeletta](https://twitter.com/timveletta)). If you also want to know how to scaffold your build pipelines using CDK, my colleague JK Gunnink [wrote a great article on it](https://mechanicalrock.github.io/2020/01/31/inception-pipeline-cdk.html). At [Mechanical Rock](https://www.mechanicalrock.io/), we love helping clients build scalable, dependable and reusable systems using infrastructure-as-code to achieve their business goals. If you think we can help you, feel free to [contact us](https://www.mechanicalrock.io/lets-get-started).
+
+## Troubleshooting
+
+I ran into a couple of issues in putting together this template and couldn't find a lot of information on how to fix them. I hope this is useful if you are running into similar issues.
+
+#### Failed to create resource. Cannot request more certificates in this account. Contact Customer Service for details.
```

Comment:
Nitpicky, but the markdown standard is to use #'s in an ordered way.

Here, you've gone from ## to ####
This should be a ###.

Realising you may have wanted to have it smaller for styling purposes or what looked "good". It may not translate well if the site/theme ever gets a redo. Or whatever.

Reply:
Good point!